salesforcecli/plugin-deploy-retrieve
# feat: more logging when progress disabled, separate test info, error …

> requires https://github.com/oclif/core/actions/runs/5244564326 and https://github.com/salesforcecli/sf-plugins-core/pull/324 to allow the noTTYOutput to be set.  fyi: oclif uses https://github.com/npkgz/cli-progress

### What does this PR do?

modify the deploy progress bar so that, when `SF_USE_PROGRESS_BAR=false` the output will 
- not include the bar graphics
- go to stderr

Also, modifies output shown by the bar to include test completion and errors, closer to what Progress would show in plugin-source.

output when SF_USE_PROGRESS_BAR=true or isn't set
```
Status: Succeeded | ████████████████████████████████████████ | 91/91 Components (Errors:0) | 11/11 Tests (Errors:0)
```

output when SF_USE_PROGRESS_BAR=false
```
Status: Pending | 0/0 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: Pending | 0/0 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: Pending | 0/0 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 6/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 6/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 22/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 24/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 64/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 74/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 77/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 88/91 Components (Errors:0) | 0/0 Tests (Errors:0)
Status: In Progress | 91/91 Components (Errors:0) | 3/11 Tests (Errors:0)
Status: In Progress | 91/91 Components (Errors:0) | 7/11 Tests (Errors:0)
Status: In Progress | 91/91 Components (Errors:0) | 10/11 Tests (Errors:0)
Status: In Progress | 91/91 Components (Errors:0) | 10/11 Tests (Errors:0)
Status: Succeeded | 91/91 Components (Errors:0) | 11/11 Tests (Errors:0)
```

### What issues does this PR fix or reference?
[@W-13131360@](https://gus.lightning.force.com/a07EE00001QXN1iYAH)
https://github.com/forcedotcom/cli/issues/2103

### QA
this can be handy to see what's happening with the streams
`SF_USE_PROGRESS_BAR=false ../plugin-deploy-retrieve/bin/dev project deploy start -d force-app --test-level RunAllTestsInOrg > stdout.txt 2> stderr.txt`
